### PR TITLE
Mention seccomp annotation removal in v1.25

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:
This patch adds removal warnings for the deprecated seccomp annotation, which is targeting removal in v1.25.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:
Refers to https://github.com/kubernetes/enhancements/issues/135#issuecomment-897670967
#### Does this PR introduce a user-facing change?

```release-note-action-required
Action required: support for the seccomp annotations `seccomp.security.alpha.kubernetes.io/pod` and `container.seccomp.security.alpha.kubernetes.io/[name]`, deprecated since 1.19, will be dropped in 1.25. Transition to using the `seccompProfile` API field.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
KEP: https://github.com/kubernetes/enhancements/blob/cccc6fa/keps/sig-node/135-seccomp/README.md
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
/sig node
/priority important-soon

PTAL @kubernetes/api-reviewers @pjbgf @tallclair

**Review by pjbgf — APPROVED** [review]
/lgtm

File: pkg/api/pod/warnings.go [hunk 1]
```diff
@@ -237,15 +237,15 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
 	// use of pod seccomp annotation without accompanying field
 	if podSpec.SecurityContext == nil || podSpec.SecurityContext.SeccompProfile == nil {
 		if _, exists := meta.Annotations[api.SeccompPodAnnotationKey]; exists {
-			warnings = append(warnings, fmt.Sprintf(`%s: deprecated since v1.19; use the "seccompProfile" field instead`, fieldPath.Child("metadata", "annotations").Key(api.SeccompPodAnnotationKey)))
+			warnings = append(warnings, fmt.Sprintf(`%s: deprecated since v1.19, removal is planned in v1.25; use the "seccompProfile" field instead`, fieldPath.Child("metadata", "annotations").Key(api.SeccompPodAnnotationKey)))
```
Comment (liggitt) [inline pkg/api/pod/warnings.go:240]: would "non-functional in v1.25+" be clearer? we won't be removing the annotations, but they will stop doing things

Comment (saschagrunert) [reply pkg/api/pod/warnings.go]: Yep, changed as suggested :+1:

**Comment by pacoxu** [comment]
The generated changelog is not correct in https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.23.md#no-really-you-must-read-this-before-you-upgrade. 

Not sure why.

**Pull request merged**